### ISSUE TITLE
Conjure Parse error on Windows for Conjure School and all .fnl files

### ISSUE THREAD

**Issue by mwebb387** [Olical/conjure#224]
Hi, I am excided to use Aniseed and Conjure to migrate my NeoVim configuration, but I am running into an issue where Conjure just does not seem to work in `Windows`.

If I run `:ConjureSchool` and then attempt to evaluate the buffer as instructed I get the following error:

```
    ; eval (buf): C:\Users\mwebb\AppData\Local\nvim\fnl\conjure-school.fnl
    ; Parse error in C:\Users\mwebb\AppData\Local\nvim\fnl\conjure-school.fnl:1
    ;   Invalid string: "C:\Users\mwebb\AppData\Local\nvim\fnl\conjure-school.fnl"
```

I get a similar error when attempting to evaluate the buffer in a test file. Here is the source of the test file:

```
(module test
  {require {a aniseed.core
            nvim aniseed.nvim}})

(a.println "Hello, World!")
```

Here is the error I get:

```
; eval (buf): C:\Users\mwebb\AppData\Local\nvim\fnl\test.fnl
; Parse error in C:\Users\mwebb\AppData\Local\nvim\fnl\test.fnl:1
;   Invalid string: "C:\Users\mwebb\AppData\Local\nvim\fnl\test.fnl"
;
; (module test
; -------------------------------------------------------------^
;
```

Any idea how to diagnose or fix this? Could this be an issue with `\` characters in Windows paths?

**Comment by zetashift** [comment]
Maaaybbbee related? https://github.com/Olical/magic-kit/issues/3

**Comment by lc9er** [comment]
I'm getting the same error with fennel files on Windows.

**Comment by zetashift** [comment]
Might be because https://github.com/Olical/conjure/blob/master/fnl/conjure/fs.fnl assumes Unix paths? Could also be in other parts of the codebase.

**Comment by lc9er** [comment]
A workaround is to use the Fennel stdio client. Steps to resolve:

1. Download Fennel binary.
2. Add it to your PATH.
3. Update your init.vim to use the [Fennel stdio option](https://github.com/Olical/conjure/wiki/Quick-start:-Fennel-(stdio)), rather than the default Aniseed client.

**Comment by mwebb387** [comment]
> A workaround is to use the Fennel stdio client

True. This works initially, but at mentioned in that link/docs:

> Warning! This is vanilla Fennel running in a regular Lua process. Unlike conjure.client.fennel.aniseed (the default) it does not come with Aniseed and it’s module macros. You’ll need to add your own standard library of choice, if you really need Aniseed’s macros then we can discuss this addition in an issue. Thanks!

I was hoping to use Aniseed and Conjure together.

I will try the above option when I get the opportunity and let you know if it works for a temporary workaround.

**Comment by mwebb387** [comment]
Confirmed that this works as a temporary workaround, but I would still like to use Aniseed along with Conjure, which might require a fix in the codebase.

I will also test when I get the chance to see if there are any issues running in WSL. If it works fine there then it very well might be a path separator issue.

**Comment by lc9er** [comment]
Agreed on both parts. I can confirm that it’s working in WSL.

**Comment by Olical** [comment]
I've just pushed a change to `develop` that uses OS aware path separators in place of the hard coded forward slashes, hopefully this helps but there may be more spots to replace.  I'll keep looking, but if anyone could try this on a windows machine and let me know if it's any better that'd be fantastic!

**Comment by Olical** [comment]
I've also pushed a fix to Aniseed and Conjure around the `*file*` variable that's defined for you. I think I needed to ensure backslashes were escaped there, which I think was the source of the original error really.

**Comment by lc9er** [comment]
> 
> 
> I've just pushed a change to `develop` that uses OS aware path separators in place of the hard coded forward slashes, hopefully this helps but there may be more spots to replace. I'll keep looking, but if anyone could try this on a windows machine and let me know if it's any better that'd be fantastic!

The `develop` branch is working for me on Windows. I'm no longer seeing errors with Fennel. Thank you Oliver.

**Comment by Olical** [comment]
So all Conjure / Aniseed + Windows things are working okay? I got it on the first try? 👀 

I'll be genuinely shocked if so, but I'll take it. I'll get it deployed soon if there's no complaints over the next few days.

**Comment by lc9er** [comment]
As far as I can tell. Fennel evaluation is working in Conjure and a plugin
that utilizes Aniseed are both working for me.

**Comment by Olical** [comment]
Should be fixed in `v4.22.0`!
